tests/lib/external/snapd-testing-tools/utils/log-filter

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+
+import argparse
+import io
+import re
+import sys
+
+import log_helper as helper
+
+# Rules are applied to filter content in the following operation details:
+# ERROR: The error shoes the bash output when a task fails
+# DEBUG: The debug includes the output of the debug script for task/suite errors
+# WARNING: The last lines from the task execution output
+# FAILED: Include a list of failed tests in the results section
+SUPPORTED_RULES = [helper.Result.FAILED.value] + helper.ExecutionInfo.list()
+
+
+def print_line(line: str):
+    if not line:
+        print()
+    else:
+        print(line.strip())
+
+
+def write_line(line: str, to_stream: io.TextIOWrapper):
+    if not line:
+        print('\n', file=to_stream)
+    else:
+        print(line.strip(), file=to_stream, end='\n')
+
+
+def compile_rules(rules, operation):
+    patterns = []
+    regex_list = []
+
+    for rule in rules:
+        parts = rule.split("=", 1)
+        if len(parts) != 2:
+            raise ValueError(
+                "Error: Rule '{}' does not follow the OPERATION=PATTERN format".format(
+                    rule
+                )
+            )
+
+        rule_operation = parts[0]
+        rule_pattern = parts[1]
+
+        if rule_operation not in SUPPORTED_RULES:
+            raise ValueError(
+                "Error: Rule operation '{}' not in supported list: {}".format(
+                    rule_operation, SUPPORTED_RULES
+                )
+            )
+
+        if operation == rule_operation:
+            patterns.append(rule_pattern)
+
+    for pattern in patterns:
+        regex_list.append(re.compile(pattern))
```

Comment:
@zyga wrote: There's no need to create another list. Construct the regex_list in the logic above.